native/cocos/platform/win32/modules/SystemWindow.cpp

```diff
@@ -64,23 +62,36 @@ bool SystemWindow::createWindow(const char *title,
 bool SystemWindow::createWindow(const char *title,
                                 int x, int y, int w,
                                 int h, int flags) {
-    _sdl->createWindow(title, x, y, w, h, flags);
+    _window = SDLHelper::createWindow(title, x, y, w, h, flags);
+    if (!_window) {
+        return false;
+    }
+
     _width = w;
     _height = h;
     return true;
 }
+
 void SystemWindow::closeWindow() {
     HWND windowHandle = reinterpret_cast<HWND>(getWindowHandle());
     if (windowHandle != 0) {
         ::SendMessageA(windowHandle, WM_CLOSE, 0, 0);
     }
 }
+
+uint32_t SystemWindow::getWindowId() const {
+    return _windowId;
+}
+
 uintptr_t SystemWindow::getWindowHandle() const {
-    return _sdl->getWindowHandle();
+    if (_externalHandle) {
+        return reinterpret_cast<uintptr_t>(_externalHandle);
+    }
+    return SDLHelper::getWindowHandle(_window);
```

Comment:
Why not just use `_windowHandle` to cache external window handle or self window handle? Then can just return `_windowHandle`.